### PR TITLE
Added source map tests

### PR DESCRIPTION
**Pull request by pksunkara**
This PR does the following:

1. Adds source map tests for small components.
2. Adds source maps for relation and assets.

### CODE REVIEW

**Comment by pksunkara** [comment]
There seems to be an issue with appveyor updating the PR status even though the tests are passing.

File: src/RefractAPI.cc [hunk 1]
```diff
@@ -342,14 +340,13 @@ namespace drafter {
         element->meta[SerializeKey::Title] = PrimitiveToRefract(MAKE_NODE_INFO(action, name));
 
         if (!action.node.relation.str.empty()) {
-            // FIXME: add SourceMap
-            element->attributes[SerializeKey::Relation] = refract::IElement::Create(action.node.relation.str);
+            refract::StringElement* relation = refract::IElement::Create(action.node.relation.str);
```
Comment (klokane) [inline src/RefractAPI.cc]: Please add explaining comment why is used combo `Element::Create()` and `attachSourceMap()` 
instead of `PrimitiveToRefract`

**Comment by pksunkara** [comment]
@klokane Is the review finished? Please mention here when it is so that I could fix the changes.

**Comment by klokane** [comment]
Yes just pleas add explaining comment as i write above

**Comment by pksunkara** [comment]
@klokane This is ready for merge. Appveyor is having problems updating the status of the PR.

**Comment by klokane** [comment]
ok

**Pull request merged**